libcache.go

```diff
@@ -0,0 +1,72 @@
+package cache
+
+import (
+	"context"
+	"time"
+
+	"github.com/shaj13/libcache"
+)
+
+var _ SimpleCache = (*TypedLibCache[string, string])(nil)
+
+type TypedLibCache[K comparable, T any] struct {
+	// caveat:
+	// libcache won't evict upon timeout, eviction only happens when it's expired + the cache key is read
+	cache libcache.Cache
+}
+
+func NewTypedLibCache[K comparable, T any](cache libcache.Cache, ttl time.Duration) *TypedLibCache[K, T] {
+	cache.SetTTL(ttl)
+	return &TypedLibCache[K, T]{
+		cache: cache,
+	}
+}
+
+func (l *TypedLibCache[K, T]) SetWithTTL(_ context.Context, key interface{}, value interface{}, ttl time.Duration) (err error) {
```

Comment:
if possible, we should either use `any` or `interface{}` but not both for consistency. What do you think?

Reply:
I’ll try with any, should work